src/main/java/frc/robot/subsystems/TCS3200_ColorSensor.java

```diff
@@ -0,0 +1,200 @@
+package frc.robot.subsystems;
+
+import edu.wpi.first.util.sendable.SendableBuilder;
+import edu.wpi.first.wpilibj.Counter;
+import edu.wpi.first.wpilibj.DigitalInput;
+import edu.wpi.first.wpilibj.DigitalOutput;
+import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+
+public class TCS3200_ColorSensor extends SubsystemBase {
+  // Output managment
+  private Counter m_counter;
+
+  // Configurable registers
+  private DigitalOutput m_s0;
+  private DigitalOutput m_s1;
+  private DigitalOutput m_s2;
+  private DigitalOutput m_s3;
+
+  private FreqScaling m_freqScaling;
+  private ColorSelect m_currColor;
+
+  private double m_lastTimeStamp;
+
+  public enum FreqScaling {
+    POWER_DOWN(0),
+    PERCENT2(1),
+    PERCENT20(2),
+    PERCENT100(3);
+
+    public final int val;
+
+    FreqScaling(int i) {
+      val = i;
+    }
+  }
+
+  public enum ColorSelect {
+    RED(0),
+    GREEN(1),
+    BLUE(2),
+    CLEAR(3);
+
+    public final int val;
+
+    ColorSelect(int i) {
+      val = i;
+    }
+  }
+
+  private double m_red;
+  private double m_green;
+  private double m_blue;
+  private double m_clear;
+
```

Comment:
The if statements cannot be the most efficient way to structure this, we also want default values for all of them